pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -58,44 +67,69 @@ func (s *Store) Signatures(ctx context.Context, name string, digest string, fn s
 		return done, err
 	}
 
-	stores := make([]store.Store, 0, len(uris))
-	for i := range uris {
-		uri := *uris[i]
-		stores = append(stores, &sigstore.Store{
-			URI:        &uri,
-			HTTPClient: s.HTTPClient,
-		})
+	var errs []error
+	stores := make([]store.Store, 0, len(customStores))
+	for _, customStore := range customStores {
+		uri, err := url.Parse(customStore.URL)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse the ClusterVersion spec.signatureStores %w", err))
+			continue
+		}
+		newHttpClient, err := s.HTTPClient(customStore.CA.Name)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to process the ClusterVersion spec.signatureStores %w", err))
+			continue
+		}
+		if customStore.CA.Name != "" {
+			stores = append(stores, &sigstore.Store{
+				URI:        uri,
+				HTTPClient: func() (*http.Client, error) { return newHttpClient, nil }})
+		}
+
+	}
+	if len(stores) == 0 {
+		return utilerrors.NewAggregate(errs)
 	}
 	store := &parallel.Store{Stores: stores}
-	if err := store.Signatures(ctx, name, digest, wrapper); err != nil || allDone {
-		return err
+	if err := store.Signatures(ctx, name, digest, wrapper); allDone {
+		if len(errs) > 0 {
+			klog.V(2).Infof("%s", utilerrors.NewAggregate(errs))
+		}
+		return nil
+	} else if err != nil {
+		errs = append(errs, err)
+		return utilerrors.NewAggregate(errs)
 	}
-	return errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature.")
+
+	errs = append(errs, errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature"))
+	return utilerrors.NewAggregate(errs)
 }
 
-func (s *Store) refreshConfiguration(ctx context.Context) ([]*url.URL, error) {
-	config, err := s.Lister.Get(s.Name)
+// refreshConfiguration retrieves the latest configuration from the ClusterVersionLister
+// and updates the customStores with the URL and CA information from the retrieved configuration.
+// It returns the updated customStores slice and any error encountered during the retrieval process.
+func (s *Store) refreshConfiguration(ctx context.Context) ([]configv1.SignatureStore, error) {
```

Comment:
If we change its signature anyway lets remove the `ctx` parameter, it is not used.

Reply:
Fixed in https://github.com/openshift/cluster-version-operator/commit/50c8a31debcbed5f1495f438d85218d0325b2ba1